### PR TITLE
Set managed table's credential to table scan/write operator's options

### PR DESCRIPTION
**Pull request by cloud-fan**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

To support credential, the first step is to get credential of the managed table and put it into table scan/write operator's options, so that it will be part of the Hadoop conf that is used to do the actual file operations.

### CODE REVIEW

File: connectors/spark/etc/conf/server.properties
Comment (ravivj-db) [inline connectors/spark/etc/conf/server.properties:1]: Given this for tests, can we move it inside the test directory ?

**Review by ravivj-db — APPROVED** [review]

**Review by jaceklaskowski — APPROVED** [review]
LGTM

File: connectors/spark/etc/conf/server.properties
Comment (cloud-fan) [reply connectors/spark/etc/conf/server.properties:1]: due to the implementation of how we load property files in the UC server, I have to put it here. We should improve the implementation to make it more testable. We can follow some examples in Spark.

File: connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java [hunk 1]
```diff
@@ -116,6 +108,66 @@ public void testDeltaPathTable() throws IOException, ApiException {
         session.stop();
     }
 
+    @Test
+    public void testManagedParquetTable() throws ApiException, IOException, ParseException {
```
Comment (tdas) [inline connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:112]: What does this have to do with "managed" table? This is only testing whether credentials vended by UC are percolated down to the relation.options. nothing more. It could be for managed tables or external tables. Table type does not matter. So the name of this test should be testCredentialUsage.

File: connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java [hunk 2]
```diff
@@ -116,6 +108,66 @@ public void testDeltaPathTable() throws IOException, ApiException {
...
+        SparkSession session = createSparkSessionWithCatalogs(SPARK_CATALOG);
+        // Spark only allow `spark_catalog` to return built-in file source tables.
+        setupManagedParquetTable(PARQUET_TABLE, new ArrayList<>(0), session);
+        String tableFullName = SPARK_CATALOG + "." + SCHEMA_NAME + "." + PARQUET_TABLE;
+
+        Dataset<Row> df = session.sql("SELECT * FROM " + tableFullName);
+        FileSourceScanExec scan = (FileSourceScanExec) df.queryExecution().executedPlan().collectLeaves().head();
+        Assert.assertEquals("test1", scan.relation().options().apply("fs.s3a.access.key"));
+        Assert.assertEquals("test2", scan.relation().options().apply("fs.s3a.secret.key"));
+        Assert.assertEquals("test3", scan.relation().options().apply("fs.s3a.session.token"));
```
Comment (tdas) [inline connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:123]: This is not really testing whether all file system access are going to use these tokens. 
I suggest a different approach completely. 
- Make UC vend different credentials (fake ones) for different paths. 
- Create a fake FileSystem object that is used for S3 paths, but actually it's a wrapper for LocalFileSystem. With that you can intercept all of file system operations that Spark will do (including file listing, etc.)
- For each of those operations, throw error if operations on path prefix X is not configured with credentials for path X. 

If the spark read operation passes, then we will have confirmed that all file system operations have been correctly configured to use the right s3 credentials. 

Does that make sense?

Comment (cloud-fan) [reply connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:123]: We probably need to update the UC sever to make it more testable. Right now the managed table location must starts with `s3://` so I can't really use a fake FileSystem to test it. Can we allow managed table location to be local file system for the ease of testing?

File: connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java (hunk 1 above)
Comment (cloud-fan) [reply connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java:112]: IIUC only managed table can have credentials? BTW I really want to test the real `CREATE TABLE` statement but UC doesn't support it today. This test has to use hacky way to create managed tables in the UC server.

**Review by haogang — APPROVED** [review]
